cloud/shared/variable_definitions.json

```diff
@@ -60,7 +60,7 @@
     "secret": false,
     "tfvar": true,
     "type": "enum",
-    "values": ["prod", "staging", "dev", "test"]
+    "values": ["prod", "staging", "ci", "dev", "test"]
```

Comment:
We should write down somewhere what is the difference between all these modes. For example I don't understand now why we need dev when we can use staging. The only thing that `dev` does right now, it seems, is that it stores terraform state locally instead of s3. Which doesn't seem to be that useful, at least to me. 

And test is specific to azure and just produces plan without actually running, right?

Reply:
yeah, I've been thinking about that too. Not sure where is the best place to write this down. as of right now:
prod - will set some things to be indestructible, use for real prod
staging - same as prod, but easier to destroy if needed
ci - skip confirmation checks
dev - store tfstate locally
test - validate things and exit early

I could alternatively just have prod and prod/staging/test here and add new flags for auto-approve and local-state.

Reply:
I'd vote for having new flag as you suggested. It will make it more explicit.

Reply:
done